### PR TITLE
Cleanup of circular buffering pass: add `currentLoadStage`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Similar to https://github.com/NVIDIA/Fuser/pull/3212

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: csrc/device_lower/pass/circular_buffer.cpp [hunk 1]
```diff
@@ -423,6 +424,24 @@ class CloneTmaCircularBufferLoopAndInsertSync
         result, for_loop_stack_);
   }
 
+  // Current load stage (for main loop): (loop_index + (stages-1)) % stages
+  Val* currentLoadStage() const {
```
Comment (rdspring1) [inline csrc/device_lower/pass/circular_buffer.cpp:428]: If `currentLoadStage` is only for the main loop, should you check for it.
```cpp
NVF_ERROR(loop_type_ == CircularBufferLoopStage::Main);
```

**Review by rdspring1 — APPROVED** [review]
LGTM.

**Comment by zasdfgbnm** [comment]
!build

**Pull request merged**